modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/internal/upgrade/v5_24_0/CPConfigurationUpgradeProcess.java

```diff
@@ -0,0 +1,187 @@
+ */
+
+package com.liferay.commerce.product.internal.upgrade.v5_24_0;
+
+import com.liferay.commerce.product.model.CPConfigurationList;
+import com.liferay.commerce.product.model.CPDefinition;
+import com.liferay.commerce.product.model.CommerceCatalog;
+import com.liferay.commerce.product.service.CPConfigurationEntryLocalService;
+import com.liferay.commerce.product.service.CPConfigurationListLocalService;
+import com.liferay.petra.string.StringBundler;
+import com.liferay.portal.kernel.language.Language;
+import com.liferay.portal.kernel.service.ClassNameLocalService;
+import com.liferay.portal.kernel.service.ClassNameLocalServiceUtil;
+import com.liferay.portal.kernel.upgrade.UpgradeProcess;
+import com.liferay.portal.kernel.util.CalendarFactoryUtil;
+import com.liferay.portal.kernel.util.GetterUtil;
+import com.liferay.portal.kernel.util.LocaleUtil;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+
+import java.util.Calendar;
+import java.util.Date;
+
+/**
+ * @author Matyas Wollner
+ */
+public class CPConfigurationUpgradeProcess extends UpgradeProcess {
+
+	public CPConfigurationUpgradeProcess(
+		ClassNameLocalService classNameLocalService,
+		CPConfigurationEntryLocalService cpConfigurationEntryLocalService,
+		CPConfigurationListLocalService cpConfigurationListLocalService,
+		Language language) {
+
+		_classNameLocalService = classNameLocalService;
+		_cpConfigurationEntryLocalService = cpConfigurationEntryLocalService;
+		_cpConfigurationListLocalService = cpConfigurationListLocalService;
+		_language = language;
+	}
+
+	@Override
+	protected void doUpgrade() throws Exception {
+		try (PreparedStatement configurationListPreparedStatement =
+				connection.prepareStatement(
+					StringBundler.concat(
+						"select CommerceCatalog.name, CommerceCatalog.userId, ",
+						"CommerceCatalog.catalogDefaultLanguageId, ",
+						"Group_.groupId, ",
+						"CPConfigurationList.CpConfigurationListId from ",
+						"CommerceCatalog join Group_ on ",
+						"CommerceCatalog.commerceCatalogId=group_.classPK and ",
+						"Group_.classNameId=? left join CPConfigurationList ",
+						"on Group_.groupId=CPConfigurationList.groupId and ",
+						"masterCPConfigurationList= ?"));
+			PreparedStatement configurationEntryPreparedStatement =
+				connection.prepareStatement(
+					StringBundler.concat(
+						"select CPDefinition.CPDefinitionId, ",
+						"CPDefinitionInventory.allowedOrderQuantities, ",
+						"CPDefinitionInventory.backOrders, ",
+						"CpDefinitionInventory.CPDefinitionInventoryEngine, ",
+						"CpDefinitionInventory.displayAvailability, ",
+						"CPDefinitionInventory.displayStockQuantity, ",
+						"CpDefinitionInventory.lowStockActivity, ",
+						"CpDefinitionInventory.maxOrderQuantity, ",
+						"CpDefinitionInventory.minOrderQuantity, ",
+						"CpDefinitionInventory.minStockQuantity, ",
+						"CpDefinitionInventory.multipleOrderQuantity, ",
+						"CPDAvailabilityEstimate.",
```

Comment:
This could come 1st due to abc sorting

Reply:
this can be left like it is because the P is uppercase in this case